### PR TITLE
Add constant folding to hetero to avoid dynamism on GPU (#10572

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - After this PR we loose dynamic capabilities for HETERO plugin

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: .ci/azure/linux.yml [hunk 1]
```diff
@@ -241,22 +241,6 @@ jobs:
...
-    # Skip test_onnx/test_zoo_models and test_onnx/test_backend due to long execution time
-  - script: |
-      export DATA_PATH=$(MODELS_PATH)
-      export MODELS_PATH=$(MODELS_PATH)
-      . $(SETUPVARS) -pyver 3.8 && python3 -m pytest -s $(INSTALL_TEST_DIR)/pyopenvino $(PYTHON_STATIC_ARGS) --junitxml=TEST-Pyngraph.xml --ignore=$(INSTALL_TEST_DIR)/pyopenvino/tests/test_utils/test_utils.py --ignore=$(INSTALL_TEST_DIR)/pyopenvino/tests/test_onnx/test_zoo_models.py --ignore=$(INSTALL_TEST_DIR)/pyopenvino/tests/test_onnx/test_backend.py
-    displayName: 'Python API 2.0 Tests'
-    continueOnError: false
-
   - script: |
       export MO_ROOT=$(INSTALL_DIR)/tools/mo
```
Comment (ilya-lavrenov) [inline .ci/azure/linux.yml:245]: why is this file changed?

Comment (ilyachur) [reply .ci/azure/linux.yml:245]: By mistake :D

**Review by ilya-lavrenov — APPROVED** [review]
